js/ui/control/geolocate_control.js

```diff
@@ -1,18 +1,57 @@
 'use strict';
 
 const Evented = require('../../util/evented');
-const browser = require('../../util/browser');
 const DOM = require('../../util/dom');
 const window = require('../../util/window');
 const util = require('../../util/util');
 
 const geoOptions = { enableHighAccuracy: false, timeout: 6000 /* 6sec */ };
 const className = 'mapboxgl-ctrl';
 
+let supportsGeolocation;
+
+function checkGeolocationSupport(callback) {
+
+    if (supportsGeolocation !== undefined) {
+        if (supportsGeolocation === true) {
+            callback();
+        }
+        return;
+    }
+
+    // Test for the case where a browser disables Geolocation because of an
+    // insecure origin
+    // https://sites.google.com/a/chromium.org/dev/Home/chromium-security/deprecating-powerful-features-on-insecure-origins
+    //
+    // navigator.permissions has incomplete browser support
+    // http://caniuse.com/#feat=permissions-api
+    if (window.navigator.permissions !== undefined) {
+        window.navigator.permissions
+            .query({ name:'geolocation' })
+            .then((p) => {
+                supportsGeolocation = p.state !== 'denied';
+                if (supportsGeolocation === true) {
+                    callback();
+                }
+            });
+    } else {
+        supportsGeolocation = !!window.navigator.geolocation;
+        if (supportsGeolocation === true) {
+            callback();
+        }
+    }
+}
+
 /**
  * A `GeolocateControl` control provides a button that uses the browser's geolocation
  * API to locate the user on the map.
  *
+ * Not all browsers support geolocation,
+ * and some users may disable the feature. Geolocation support for modern
+ * browsers including Chrome requires sites to be served over HTTPS. If
+ * geolocation support is not available, the GeolocateControl will not
+ * be visible.
+ *
```

Comment:
I don't use wrapping too, it just that 70 seems too short, given that we use up to 120 for the code.